python/lsst/dax/imgserv/jobqueue/imageworker.py

```diff
@@ -78,17 +80,30 @@ def get_image_async(self, *args, **kwargs):
     config["DAX_IMG_CONFIG"] = config_path
     meta_url = imgserv_config.webserv_config["dax.imgserv.meta.url"]
     config["DAX_IMG_META_URL"] = meta_url
-    soda = imageSODA.ImageSODA(config)
-    image = soda.do_sync(params)
-    # save the result image in local temp dir
-    with tempfile.NamedTemporaryFile(dir=config["DAX_IMG_TEMPDIR"],
+    image = None; status = "OK"
+    try:
+        soda = imageSODA.ImageSODA(config)
+        image = soda.do_sync(params)
+    except ImageNotFound:
+        status = "Error=Image Not Found"
+    except UsageError as e:
+        status = "Error="+e.message
+    except Exception as e:
+        status = str(e)
```

Comment:
Okay, so this error path is really different from the other error pathways in the image_soda more flasky parts.  I would suggest doing the same things that the unhandled exception is doing to catch the different exceptions and format the messages.  (For example, you probably want to return the callstack of the exception, if you can).  If you can tie in both of these paths to generate the error message, that would be much more consistent and give a lot more info here.  And I agree with Tim, don't override the Image Not Found exception error message - in fact, use whatever is in there, and always have a useful message when raising those exceptions.

Reply:
All these gone, now simpler with using the default exception handling by Celery.